### PR TITLE
feat(s42): LLM playtester agent harness

### PR DESCRIPTION
**Pull request by theinterneti**
## Summary

Implements [S42 — LLM Playtester Agent Harness](specs/42-llm-playtester-agent-harness.md).

- **`PlaytesterAgent`** with `setup()/run()` lifecycle; deterministic via `run_seed` + `persona_jitter_seed`
- **`TasteProfile`** frozen dataclass + 5 built-in personas (curious-explorer, disengaged-skeptic, impulsive-actor, terse-minimalist, verbose-narrator) with configurable jitter
- **`PlaytestReport`** / `Commentary` / `TurnRecord` Pydantic models
- `MAX_CONSECUTIVE_TIMEOUTS=3` abandonment logic (AC-42.05)
- Verbosity brevity constraint: verbosity ≤ 0.1 → '8 words' system hint (AC-42.04)
- JSON commentary parsing with markdown-fence stripping and safe fallback

## Tests
- 30 new unit tests: 25 pure-function edge-case tests + 5 `@pytest.mark.spec` AC compliance tests
- AC-42.01 through AC-42.05 all covered
- 2649 passed, 86.76% coverage

## AC Traceability
- AC-42.01 ✅ Complete session (≥ PLAYTEST_MIN_TURNS turns)
- AC-42.02 ✅ Commentary bounds per turn
- AC-42.03 ✅ Reproducibility (same seeds → same inputs)
- AC-42.04 ✅ Verbosity brevity constraint in system prompt
- AC-42.05 ✅ Consecutive timeout abandonment

### CODE REVIEW

File: src/tta/playtest/agent.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+    try:
+        data = json.loads(cleaned)
+        return Commentary(
+            turn_index=turn_index,
+            agent_intent=str(data.get("agent_intent", "")),
+            surprise_level=float(data.get("surprise_level", 0.5)),
+            surprise_note=str(data.get("surprise_note", "")),
+            coherence_rating=float(data.get("coherence_rating", 0.5)),
+            coherence_note=str(data.get("coherence_note", "")),
+        )
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/playtest/agent.py:373]: Fixed.

File: src/tta/playtest/agent.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+    FR-42.07 — Stateless: no shared state between parallel agents.
+    """
+
+    def __init__(
+        self,
+        api_base_url: str,
+        llm_client: LLMClient,
+        llm_model: str = "gpt-4o-mini",
+        api_key: str | None = None,
+    ) -> None:
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/playtest/agent.py:44]: Fixed.

File: src/tta/seeds/validator.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    def _check_id(self, raw: dict[str, Any], path: Path) -> None:
+        seed_id = str(raw.get("id", ""))
+        if not _ID_PATTERN.match(seed_id):
+            raise SeedSchemaError(
+                f"Seed at {path}: id {seed_id!r} must match [a-z0-9-]+"
+            )
+        if len(seed_id) > _MAX_ID_LEN:
+            raise SeedSchemaError(
+                f"Seed at {path}: id must be ≤{_MAX_ID_LEN} chars, got {len(seed_id)}"
+            )
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/seeds/validator.py:104]: Fixed.

File: src/tta/genesis/genesis_v2.py [hunk 4]
```diff
@@ -31,8 +31,46 @@
...
+    if not seed_id:
+        return
+    manifest = registry.get(seed_id)
+    if manifest is None:
+        log.warning("apply_seed_composition_not_found", seed_id=seed_id)
+        return
+    comp_dict = manifest.composition.to_dict()
+    comp_dict["seed_id"] = manifest.id
+    comp_dict["seed_version"] = manifest.version
+    config["composition"] = comp_dict
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/genesis/genesis_v2.py:69]: Fixed.

File: tests/unit/playtest/conftest.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+_GAME_ID = "game-abc-123"
+_INTRO_NARRATIVE = "You stand at a misty bus stop. Rain taps the shelter roof."
+
+
+def _make_http_mock(
+    game_id: str = _GAME_ID,
+    intro_narrative: str = _INTRO_NARRATIVE,
+    turns_to_serve: int = 5,
+    simulate_timeout_turns: set[int] | None = None,
+) -> MagicMock:
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply tests/unit/playtest/conftest.py:104]: Fixed.

File: tests/unit/playtest/test_s42_ac_compliance.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    agent = PlaytesterAgent(api_base_url="http://test", llm_client=mock_llm)
+    agent.setup("seed-cosmos", "curious-explorer", run_seed=42)
+    # Override profile AFTER setup() to guarantee verbosity well below 0.1 threshold.
+    agent._profile = TasteProfile(  # type: ignore[attr-defined]
+        verbosity=0.05,
+        boldness=0.3,
+        curiosity=0.4,
+        genre_affinity="any",
+        tone_affinity="neutral",
+    )
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply tests/unit/playtest/test_s42_ac_compliance.py:112]: Fixed.

File: src/tta/playtest/agent.py [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+            current_narrative = game_data.get("narrative_intro", "")
+
+            consecutive_timeouts = 0
+            turn_index = 0
+
+            while turn_index < PLAYTEST_MIN_TURNS:
+                # Consume RNG once per turn for reproducibility (FR-42.06)
+                rng_value = self._rng.random()
+                try:
+                    result = await asyncio.wait_for(
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/playtest/agent.py]: Fixed.

File: src/tta/playtest/agent.py [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+            headers["Authorization"] = f"Bearer {self._api_key}"
+
+        async with httpx.AsyncClient(
+            base_url=self._api_base_url,
+            headers=headers,
+            timeout=PLAYTEST_TURN_TIMEOUT + 10,
+        ) as client:
+            resp = await client.post(
+                "/api/v1/games",
+                json={"world_id": None, "preferences": {}},
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/playtest/agent.py]: Fixed.

File: src/tta/playtest/agent.py [hunk 9]
```diff
@@ -0,0 +1,368 @@
...
+            run_id=self._run_id,
+            run_seed=self._run_seed,
+            scenario_seed_id=self._scenario_seed_id,
+            persona_id=self._persona_id,
+            persona_jitter_seed=self._persona_jitter_seed,
+            model=self._llm_model,
+            status=status,
+            genesis_phases_completed=7,
+            gameplay_turns_completed=len(completed),
+            turns=list(self._turns),
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/playtest/agent.py:311]: Fixed.

File: src/tta/playtest/agent.py [hunk 10]
```diff
@@ -0,0 +1,368 @@
...
+            system_parts.append(
+                f"You gravitate toward: {', '.join(profile.trope_affinity)}."
+            )
+        if profile.curiosity > 0.7:
+            system_parts.append("You like to ask follow-up questions.")
+        if profile.meta_awareness > 0.7:
+            system_parts.append("You sometimes comment on game mechanics.")
+        # AC-42.04: enforce brevity constraint for very terse personas
+        if profile.verbosity <= 0.1:
+            system_parts.append("Keep your response under 8 words.")
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/playtest/agent.py]: Fixed.

File: src/tta/playtest/profile.py [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+        tropes = tuple(template.get("trope_affinity", []))
+        return cls(
+            verbosity=maybe_jitter(float(template["verbosity"])),
+            boldness=maybe_jitter(float(template["boldness"])),
+            curiosity=maybe_jitter(float(template["curiosity"])),
+            genre_affinity=template["genre_affinity"],
+            tone_affinity=template["tone_affinity"],
+            trope_affinity=tropes,
+            attention_span=maybe_jitter(float(template.get("attention_span", 0.8))),
+            meta_awareness=maybe_jitter(float(template.get("meta_awareness", 0.1))),
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/playtest/profile.py]: Fixed.

File: src/tta/seeds/registry.py [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+
+    def list(
+        self,
+        tags: list[str] | None = None,
+        genre: str | None = None,
+    ) -> list[SeedManifest]:
+        """Return seeds matching any of *tags* and/or *genre*."""
+        results = list(self._seeds.values())
+        if tags:
+            results = [s for s in results if any(t in s.tags for t in tags)]
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/seeds/registry.py]: Fixed.

File: src/tta/api/app.py [hunk 13]
```diff
@@ -220,6 +220,13 @@ async def _lifespan(app: FastAPI) -> AsyncIterator[None]:
         count=len(app.state.template_registry.list_all()),
     )
 
+    # 5c. Seed registry — loads scenario seeds (S41)
+    from tta.seeds.registry import SeedRegistry
+
+    seeds_dir = Path(__file__).resolve().parent.parent.parent.parent / "data" / "seeds"
```
[the remark this answers is not in the document]
Comment (theinterneti) [reply src/tta/api/app.py]: Fixed.
